### PR TITLE
`lexer.test.c` Initial test

### PR DESCRIPTION
**Pull request by abdorah**

### CODE REVIEW

**Review by devkadirselcuk — APPROVED** [review]

**Review by devkadirselcuk — APPROVED** [review]

**Review by devkadirselcuk — DISMISSED** [review]

File: unit-tests/lexer.test.c [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+{
+	TEST_ASSERT_EQUAL_INT(EXIT_SUCCESS, lexerCheck(lex, errors));
+}
+
+int main(void)
+{
+	UNITY_BEGIN();
+	RUN_TEST(test_function_should_check_lex);
+	return UNITY_END();
+}
```
Comment (jbampton) [inline unit-tests/lexer.test.c:35]:
```suggestion
}

```

**Review by jbampton — CHANGES_REQUESTED** [review]

**Comment by BaseMax** [comment]
**Just to be clear:**

tests directory is a bit old and out to date.
`input.one` is updated. We have to create so many new tests at `tests` directory.

**Comment by jbampton** [comment]
`pre-commit` failed and so did the `EditorConfig` check for the missing end of line which I reviewed.

Next time you do a pull request make sure to have `pre-commit` installed and working.

It saves time on code reviews.

https://github.com/One-Language/One/blob/master/CONTRIBUTING.md

**Comment by jbampton** [comment]
Screen Shot 2021-06-06 at 11 05 01 pm

**Comment by abdorah** [comment]
> **Just to be clear:**
> 
> tests directory is a bit old and out to date.
> `input.one` is updated. We have to create so many new tests at `tests` directory.

Hello @BaseMax ,  I understand what you mentioned here. Indeed, I think that we have to provide the needed files (*.one) for tests alongside with tests. My concerns first are to define the right structure of the project that allows us to add the others tests easily.
Thank you for your reviews.

**Comment by BaseMax** [comment]
Thanks Kotbi.

**Review by devkadirselcuk — APPROVED** [review]

**Comment by BaseMax** [comment]
Also Thanks from devkadirselcuk.
I going to review and check and pre-commit...

**Comment by BaseMax** [comment]
How you plan to run tests?
Do you want to write a python or bash script?
I think we can add this to CMAKE and use `make test` or something like it.

**Comment by BaseMax** [comment]
Ooops
```
[max@workstation build]$ cmake ..
-- Configuring done
-- Generating done
-- Build files have been written to: /home/max/Projects/One/build
[max@workstation build]$ make
Consolidate compiler generated dependencies of target One
[  5%] Building C object CMakeFiles/One.dir/src/arg.c.o
[ 11%] Building C object CMakeFiles/One.dir/src/array.c.o
[ 17%] Building C object CMakeFiles/One.dir/src/compile.c.o
[ 23%] Building C object CMakeFiles/One.dir/src/error.c.o
[ 29%] Building C object CMakeFiles/One.dir/src/lexer.c.o
[ 35%] Building C object CMakeFiles/One.dir/src/one.c.o
[ 41%] Building C object CMakeFiles/One.dir/src/ast.c.o
[ 47%] Building C object CMakeFiles/One.dir/src/parser.c.o
[ 52%] Building C object CMakeFiles/One.dir/src/tree.c.o
[ 58%] Building C object CMakeFiles/One.dir/src/token.c.o
[ 64%] Building C object CMakeFiles/One.dir/src/vm.c.o
[ 70%] Building C object CMakeFiles/One.dir/src/file.c.o
[ 76%] Building C object CMakeFiles/One.dir/unit-tests/unity/unity.c.o
[ 82%] Building C object CMakeFiles/One.dir/unit-tests/ast.test.c.o
[ 88%] Building C object CMakeFiles/One.dir/unit-tests/lexer.test.c.o
In file included from /home/max/Projects/One/unit-tests/lexer.test.c:2:
/home/max/Projects/One/src/error.h:33:11: error: field ‘location’ has incomplete type
   33 |  Location location;
      |           ^~~~~~~~
In file included from /home/max/Projects/One/unit-tests/lexer.test.c:3:
/home/max/Projects/One/src/lexer.h:25:11: error: field ‘location’ has incomplete type
   25 |  Location location;
      |           ^~~~~~~~
make[2]: *** [CMakeFiles/One.dir/build.make:272: CMakeFiles/One.dir/unit-tests/lexer.test.c.o] Error 1
```

**Comment by BaseMax** [comment]
Remember we need location structure and it's defined at `token.h`.
So you have to include `token.h` also at your test files!

Just let me know if you want to fix this.
Since we need this to be fixed ASAP. (fast)

**Pull request merged**